### PR TITLE
refactor(core): Introduce overload for record-type node parameter (no-changelog)

### PR DESCRIPTION
**Pull request by ivov**

### CODE REVIEW

File: packages/workflow/src/Interfaces.ts [hunk 1]
```diff
@@ -564,13 +564,23 @@ namespace ExecuteFunctions {
...
 			parameterName: 'resource',
 			itemIndex?: number,
 		): T['resource'];
 
+		getNodeParameter(
+			parameterName: RecordReturning.NodeParameter,
+			itemIndex: number,
+			fallbackValue?: IDataObject,
+			options?: IGetNodeParameterOptions,
+		): IDataObject;
```
Comment (ivov) [inline packages/workflow/src/Interfaces.ts:583]: Ideally we should narrow down `IDataObject` in future, but too widely used to refactor in this PR.

Comment (netroy) [reply packages/workflow/src/Interfaces.ts:583]: I'm certain any PR wanting to replace `IDataObject` with stronger types is going to me much larger than this PR 😮‍💨 🤦🏽

**Review by netroy — APPROVED** [review]
LGTM

**Comment by janober** [comment]
Got released with `n8n@0.204.0`

**Pull request merged**